src/server.ts

```diff
@@ -26,120 +33,170 @@ export class MyAgent extends Agent<Env> {
     apiKey: this.env.GROQ_API_KEY,
   })
   messages = RESET
+  connection!: Connection
 
   constructor(state: DurableObjectState, env: Env) {
     super(state, env)
   }
 
   onConnect(connection: Connection, ctx: ConnectionContext): void | Promise<void> {
+    this.connection = connection
     console.log('Connected to agent')
     // Clear history
     this.messages = RESET
```

Comment:
Explicitly resetting history on every new connection, cause I couldn't figure out how to change the DO instance from the frontend 🙃